actionners/gcp/function/function.go

```diff
@@ -0,0 +1,254 @@
+package functions
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"cloud.google.com/go/functions/apiv2/functionspb"
+	"google.golang.org/api/idtoken"
+
+	"github.com/falco-talon/falco-talon/internal/events"
+	"github.com/falco-talon/falco-talon/internal/gcp/checks"
+	"github.com/falco-talon/falco-talon/internal/gcp/client"
+	"github.com/falco-talon/falco-talon/internal/models"
+	"github.com/falco-talon/falco-talon/internal/rules"
+	"github.com/falco-talon/falco-talon/utils"
+)
+
+const (
+	Name          string = "function"
+	Category      string = "gcp"
+	Description   string = "Invoke a GCP function forwarding the Falco event payload"
+	Source        string = "any"
+	Continue      bool   = true
+	AllowOutput   bool   = false
+	RequireOutput bool   = false
+	Permissions   string = `{
+		"roles/cloudfunctions.invoker"
+	}`
+	Example string = `- action: Invoke GCP Cloud Function
+	actionner: gcp:function
+	parameters:
+	  gcp_function_name: sample-function
+	  gcp_function_location: us-central1
+      gcp_function_timeout: 10
+	`
+)
+
+var (
+	RequiredOutputFields = []string{}
+)
+
+type Parameters struct {
+	GCPFunctionName     string `mapstructure:"gcp_function_name" validate:"required"`
+	GCPFunctionLocation string `mapstructure:"gcp_function_location" validate:"required"`
+	GCPFunctionTimeout  int    `mapstructure:"gcp_function_timeout"`
```

Comment:
maybe adding a validation to enforce >= 0 or use a uint

Reply:
Default value would be zero, and further down we check if it's different than 0 meaning user has set otherwise:
```
	if parameters.GCPFunctionTimeout != 0 {
		httpClient := http.Client{
			Timeout: time.Duration(parameters.GCPFunctionTimeout),
		}
		c.SetHTTPClient(&httpClient)
	}
```

Reply:
`!= `0 means it could be `-1000`